Spam2

keshavsethi opened the pull request:
Fixes ##7927 
and with the feedback from PR #7928  
Note: This is big PR as it doesn't hinder master much. I have added all views in separate URL i.e. /spam2. **_If you want I can divide this in small PRs and milestones for better Review._**  

@jywarren @SidharthBansal @ananyaarun @pydevsg @emilyashley @VladimirMikulic @cesswairimu @ebarry 

**I have made some changes as per feedback from @jywarren and others**
1. **Made /spam2 as an alternate URL** so that it becomes easy for moderators and reviewers to see the changes and compare between two. 
2.  **Removed links and features that are not working** as suggested by @jywarren.now all the features in UI work except Dashboard which I made grey for reference.
3. Added some **Bulk moderation features** like bulk_spam, Bulk_publish, Select all/none, Bulk_delete, Bulk_ban. 
4. Made it  **responsive** for all devices and the table is scrollable horizontally within the card.
5. Added color for status of post as mentioned by @jywarren in #7953. I have commented there as well.
### Screenshots 
![Screenshot from 2020-05-28 01-33-13](https://user-images.githubusercontent.com/36025262/83068598-398fab00-a086-11ea-9363-588f21c9ad64.png)
![Screenshot from 2020-05-28 01-33-21](https://user-images.githubusercontent.com/36025262/83068614-3b596e80-a086-11ea-92c0-8523b18929f5.png)
![Screenshot from 2020-05-28 01-34-05](https://user-images.githubusercontent.com/36025262/83068628-3e545f00-a086-11ea-8b55-9d211f6d1074.png)
![Screenshot from 2020-05-28 01-34-15](https://user-images.githubusercontent.com/36025262/83068638-42807c80-a086-11ea-8015-c59cc68054ca.png)
### Gifs
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/36025262/83068768-70fe5780-a086-11ea-8f0e-0424ea9f0313.gif)
![ezgif com-crop (1)](https://user-images.githubusercontent.com/36025262/83068781-7360b180-a086-11ea-8f56-09a3ff4f03d8.gif)

Tests are left as of now I am working on them. I will push them soon. 
@jywarren @SidharthBansal @ananyaarun @pydevsg @emilyashley @VladimirMikulic @cesswairimu @ebarry Please give your feedbacks

_Thanks!_

VladimirMikulic commented:
As far as UI is concerned, this is a great improvement, but there are some minor issues.

Some of the items in the table are not vertically centred, which makes it look clunky.

Buttons/actions are inconsistent as well. Some buttons are rounded and some of them are rectangles with a small border-radius. Taking into account the current design system of Public Lab's website, rectangles with a small border-radius fit better.

keshavsethi commented:
> As far as UI is concerned, this is a great improvement, but there are some minor issues.
> 
> Some of the items in the table are not vertically centred, which makes it look clunky.
> 
> Buttons/actions are inconsistent as well. Some buttons are rounded and some of them are rectangles with a small border-radius. Taking into account the current design system of Public Lab's website, rectangles with a small border-radius fit better.

I have aligned table headers and text. 
![Screenshot from 2020-05-28 04-45-35](https://user-images.githubusercontent.com/36025262/83081575-a6af3a80-a09e-11ea-9fea-a7149a54b6b6.png)

Buttons are also made consistent in mobile view
![Screenshot from 2020-05-28 04-45-49](https://user-images.githubusercontent.com/36025262/83081624-be86be80-a09e-11ea-9b7a-c0f07bff7144.png)
I will change the action button  

Thanks !!

keshavsethi commented:
I have changed the UI as this looks a bit odd as compared to the website. I removed the page header and added some basic stuff. I will make another PR for this and closing this one. 
Thanks!!
